client/client.go

```diff
@@ -423,13 +425,83 @@ func (c *Client) getLocalMeta() error {
 			c.loadTargets(targets.Targets)
 		}
 	}
+
+	if loadFailed {
+		// If any of the metadata failed to be verified, return the reason for that failure
+		// and fail fast before delegated targets
+		return retErr
+	}
+
+	// verifiedDelegatedTargets is a set of verified delegated targets
+	verifiedDelegatedTargets := make(map[string]bool)
+	for fileName := range meta {
+		if roles.IsDelegatedTargetsManifest(fileName) {
```

Comment:
(optimization only) should we check if the entry isn't already in `verifiedDelegatedTargets`? In the example you have below we may check `a.json`, `b.json`, even though `c.json` gives them both as `verifiedDelegatedTargets`

Reply:
You're right, that could speed up things. Done in [cfe757e](https://github.com/theupdateframework/go-tuf/pull/384/commits/cfe757e58638c77ff23ba9e22e49255ad791b150)!